File: PY/sistema_telefonos_personas.py
# sistema_personas_telefonos.py

class SistemaPersonas:
    def __init__(self):
        self.personas = []
    
    def iniciar(self):
        print("Sistema de Registro de Personas y Teléfonos")
        self.cargar_persona()
    
    def cargar_persona(self):
        nombre = input("Nombre: ")
        apellido = input("Apellido: ")
        dni = input("DNI: ")
        telefonos = self.cargar_telefonos()
        
        self.personas.append([nombre, apellido, dni, telefonos])
        
        continuar = input("¿Desea cargar otra persona? (s/n): ").lower()
        if continuar == 's':
            self.cargar_persona()
        else:
            self.mostrar_datos()
    
    def cargar_telefonos(self):
        telefonos = []
        while True:
            telefono = input(f"Teléfono {len(telefonos) + 1} (deje vacío para terminar): ")
            if not telefono.strip():
                if len(telefonos) == 0:
                    print("Debe ingresar al menos un teléfono.")
                    continue
                else:
                    break
            telefonos.append(telefono)
        return telefonos
    
    def mostrar_datos(self):
        print("\nDatos registrados:")
        print("-----------------")
        for persona in self.personas:
            print(persona)

# Iniciar el sistema
if __name__ == "__main__":
    sistema = SistemaPersonas()
    sistema.iniciar()

File: JS/sistema_telefonos_personas.js
import rd from "readline";

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let personas = [];

function iniciarSistema() {
  console.log("Sistema de Registro de Personas y Teléfonos");
  cargarPersona();
}

function cargarPersona() {
  rl.question("Nombre: ", (nombre) => {
    rl.question("Apellido: ", (apellido) => {
      rl.question("DNI: ", (dni) => {
        const telefonos = [];
        cargarTelefonos(nombre, apellido, dni, telefonos);
      });
    });
  });
}

function cargarTelefonos(nombre, apellido, dni, telefonos) {
  rl.question(
    `Teléfono ${telefonos.length + 1} (deje vacío para terminar): `,
    (telefono) => {
      if (telefono.trim() === "") {
        if (telefonos.length === 0) {
          console.log("Debe ingresar al menos un teléfono.");
          cargarTelefonos(nombre, apellido, dni, telefonos);
        } else {
          personas.push([nombre, apellido, dni, telefonos]);
          preguntarContinuar();
        }
      } else {
        telefonos.push(telefono);
        cargarTelefonos(nombre, apellido, dni, telefonos);
      }
    }
  );
}

function preguntarContinuar() {
  rl.question("¿Desea cargar otra persona? (s/n): ", (respuesta) => {
    if (respuesta.toLowerCase() === "s") {
      cargarPersona();
    } else {
      mostrarDatos();
      rl.close();
    }
  });
}

function mostrarDatos() {
  console.log("\nDatos registrados:");
  console.log("-----------------");
  personas.forEach((persona) => {
    console.log(persona);
  });
}

// Iniciar el sistema
iniciarSistema();
